website/demos/components/DraggableHeaderRenderer.tsx

```diff
@@ -11,31 +11,52 @@ export function DraggableHeaderRenderer<R>({
   column,
   ...props
 }: DraggableHeaderRendererProps<R>) {
-  const [{ isDragging }, drag] = useDrag({
-    type: 'COLUMN_DRAG',
-    item: { key: column.key },
-    collect: (monitor) => ({
-      isDragging: monitor.isDragging()
-    })
-  });
-
-  const [{ isOver }, drop] = useDrop({
-    accept: 'COLUMN_DRAG',
-    drop({ key }: { key: string }) {
-      onColumnsReorder(key, column.key);
-    },
-    collect: (monitor) => ({
-      isOver: monitor.isOver(),
-      canDrop: monitor.canDrop()
-    })
-  });
+  const [isDragging, toggleDragging] = useState(false);
+  const [isOver, toggleIsOver] = useState(false);
+
+  function handleDragStart(event: React.DragEvent<HTMLDivElement>) {
+    event.dataTransfer.setData('text/plain', column.key);
+    event.dataTransfer.setDragImage(event.currentTarget.closest('.rdg-cell')!, 0, 0);
+    toggleDragging(true);
+  }
+
+  function handleDragEnd() {
+    toggleDragging(false);
+  }
+
+  function handleDragEnter() {
+    toggleIsOver(true);
+  }
+
+  function handleDragLeave() {
+    toggleIsOver(false);
+  }
+
+  function handleDragOver(event: React.DragEvent<HTMLDivElement>) {
+    // prevent default to allow drop
+    event.preventDefault();
+  }
+
+  function handleDrop(event: React.DragEvent<HTMLDivElement>) {
+    const sourceKey = event.dataTransfer.getData('text/plain');
+    if (sourceKey !== column.key) {
+      event.preventDefault();
+      onColumnsReorder(sourceKey, column.key);
+    }
+    toggleIsOver(false);
```

Comment:
Small tweak: we can call `toggleIsOver(false);` at the top of the function,
that way if for example `onColumnsReorder()` throws an error, then we'll at least end up with the correct state.